plugins/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -0,0 +1,148 @@
+        task_config: Configuration for the task.
+        label_column: Index of the class label column in the CSV dataset.
+    """
+
+    _TASK_TYPE = "xgboost"
+
+    def __init__(
+        self,
+        name: str,
+        inputs: Dict[str, Type],
+        hyperparameters: Optional[Dict[str, Union[str, int, float, bool]]] = None,
+        task_config: Optional[TrainParameters] = None,
+        label_column: Optional[int] = 0,
+        **kwargs,
+    ):
+        self._hyperparameters = hyperparameters
+        self._train_parameters = task_config
+        self._label_column = label_column
+
+        outputs = {
+            "model": JoblibSerializedFile,
+            "predictions": List[float],
+            "evaluation_result": Dict[str, Dict[str, List[float]]],
+        }
+
+        super(XGBoostTask, self).__init__(
+            name,
+            task_type=self._TASK_TYPE,
+            task_config=task_config,
+            interface=Interface(inputs=inputs, outputs=outputs),
+            **kwargs,
+        )
+
+    # Train method
+    def train(self, dtrain, dvalid, **kwargs) -> Tuple[str, Dict[str, Dict[str, List[float]]]]:
+        evals_result = {}
+        # if validation data is provided, then populate evals and evals_result
+        booster_model = (
+            xgboost.train(
+                params=self._hyperparameters,
+                dtrain=dtrain,
+                **asdict(self._train_parameters if self._train_parameters else TrainParameters()),
+                evals=[(dvalid, "validation")],
+                evals_result=evals_result,
+            )
+            if dvalid
+            else xgboost.train(
+                params=self._hyperparameters,
+                dtrain=dtrain,
+                **asdict(self._train_parameters if self._train_parameters else TrainParameters()),
+            )
+        )
+        fname = "booster_model.joblib.dat"
+        joblib.dump(booster_model, fname)
+        return fname, evals_result
+
+    # Test method
+    def test(self, booster_model, dtest, **kwargs) -> List[float]:
+        booster_model = joblib.load(booster_model)
+        y_pred = booster_model.predict(dtest).tolist()
+        return y_pred
+
+    def execute(self, **kwargs) -> Any:
+        train_key = ""
+        test_key = ""
+        validation_key = ""
+
+        # fetch inputs
+        for key in self.python_interface.inputs.keys():
+            if "train" in key:
+                train_key = key
+            elif "test" in key:
+                test_key = key
+            elif "valid" in key:
+                validation_key = key
+
+        if not (train_key and test_key):
+            raise ValueError("Must have train and test inputs")
+
+        dataset_vars = {}
+
+        for each_key in [train_key, test_key, validation_key]:
+            if each_key:
+                dataset = kwargs[each_key]
+                if issubclass(self.python_interface.inputs[each_key], str):
+                    if "csv" in dataset:
```

Comment:
can we make this FlyteFile, and this this should be simpler to use, you will not need to do `csv` in dataset?

Reply:
I have included both string and FlyteFile as valid types for train and test data. Do you want me to remove the string data type?